### PR TITLE
Add follow-up blog post on Akka Typed and Dotty

### PR DESCRIPTION
**Pull request by eloots**
Draft of a follow-up article.

### CODE REVIEW

File: posts/2020-02-12-using-dotty-union-types-with-akka-typed-part-II.adoc [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+  ...
+}
+----
+
+== Restoring the `Pinger` actor's protocol - Dotty version
+
+The Dotty version of the `Pinger` actor suffers from the same problem. The protocol of the actor is _widened_ with the `PingPong.Pong` message. This means that _any_ actor can send a `Pong` message to the `Pinger` actor. The solution in this case is different from previous one:
+
+* The `apply` method in the `Pinger` object, will return a `Behavior[Command]` thus restoring the original protocol
+* The `Pinger` actor defines an internal behavior of type `Behavior[CommandIncludingResponses]`
```
Comment (LeonorLunatech) [inline posts/2020-02-12-using-dotty-union-types-with-akka-typed-part-II.adoc]: In the code snipped you have renamed `CommandsAndResponses` to `CommandsAndResponses`

File: posts/2020-02-12-using-dotty-union-types-with-akka-typed-part-II.adoc [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+          context.log.info(s"Hey: I just received a $response !!!")
+          Behaviors.same
+      }
+    }
+    internalBehavior.narrow
+  }
+}
+----
+
+Admittedly, some magic has happened here and the question to ask is, _What is this `narrow` method doing?_. Let's look at that in the next section.
```
Comment (LeonorLunatech) [inline posts/2020-02-12-using-dotty-union-types-with-akka-typed-part-II.adoc]: Just and extra period `.`, after the question mark

File: posts/2020-02-12-using-dotty-union-types-with-akka-typed-part-II.adoc [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+
+First, we note that the `Behavior` class is generic: it has a type parameter `T`, which, because of no specific variance annotation on `T`, implies that `Behavior` is _invariant_ in its type parameter `T`. Also note the comment on the `narrow` method, stating: 
+
+> _This method is necessary to implement the contravariant nature of Behavior (which cannot be expressed directly due to type inference problems)_.
+
+Second, we see that the class `ExtensibleBehavior`, which is a subclass of `Behavior`, has a `receive` method which takes a parameter `msg` of type `T`. Because functions (or _methods_), are contravariant in their argument types, the only possible variance case for the type parameter is invariant (`T`) or contravariant (`-T`). Because of type inference problems in Scala 2, the former was chosen.
+
+NOTE: Variance manifests itself in specific contexts and is a topic in itself with _contravariance_ being the least intuitive. We'll see however that in the case of `Behavior`, it is actually quite easy to understand. I'll get back to that later. For a comprehensive explanation of variance in Scala read https://docs.scala-lang.org/tour/variances.html[this article].
```
Comment (LeonorLunatech) [inline posts/2020-02-12-using-dotty-union-types-with-akka-typed-part-II.adoc]: I think you need to one more comma to the sentence `We'll see however that in the case of Behavior,` meaning: `We'll see however that, in the case of Behavior,`

File: posts/2020-02-12-using-dotty-union-types-with-akka-typed-part-II.adoc (hunk 1 above)
Comment (eloots) [reply posts/2020-02-12-using-dotty-union-types-with-akka-typed-part-II.adoc]: Good catch!

File: posts/2020-02-12-using-dotty-union-types-with-akka-typed-part-II.adoc (hunk 2 above)
Comment (eloots) [reply posts/2020-02-12-using-dotty-union-types-with-akka-typed-part-II.adoc]: Good catch. Removed the '.' and moving the question mark outside the italic formatting.

File: posts/2020-02-12-using-dotty-union-types-with-akka-typed-part-II.adoc (hunk 3 above)
Comment (eloots) [reply posts/2020-02-12-using-dotty-union-types-with-akka-typed-part-II.adoc]: > I think you need to one more comma to the sentence

Correct and corrected!

**Pull request merged**